syntax = "proto3";
package cliente_gateway;

message ClienteRequisicao {
  enum TipoRequisicao {
    REQUISICAO_DESCONHECIDA = 0;
    LISTAR_DISPOSITIVOS = 1;
    ENVIAR_COMANDO = 2;
  }

  TipoRequisicao request_type = 1;
  Comando comando = 2;
}

message Comando {
  enum TipoComando {
    COMANDO_DESCONHECIDO = 0;
    ALTERNA_ENERGIA = 1;
    SETAR_TEMPERATURA = 2;
  }

  string id_despositivo = 1;
  TipoComando Comando_type = 2;
  string value = 3;
}

message RespostaGateway {
  enum TipoRequisicao {
    REQUISICAO_DESCONHECIDA = 0;
    LISTAR_DISPOSITIVOS = 1;
    ENVIAR_COMANDO = 2;
    ERRO = 3;
  }

  TipoRequisicao response_type = 1;
  bool sucesso = 2;
  string message = 3;
  repeated DispositivoInfo devices = 4;
}

message DispositivoInfo {
  string id = 1;
  string tipo = 2;
  string ip = 3;
  int32 porta = 4;
  string estado = 5;
}
